internal/pkg/service/stream/dependencies/coordinator.go

```diff
@@ -39,38 +30,36 @@ func (v *coordinatorScope) StatisticsL2Cache() *cache.L2 {
 }
 
 func NewCoordinatorScope(ctx context.Context, svcScope ServiceScope, distScp dependencies.DistributionScope, distLockScp dependencies.DistributedLockScope, cfg config.Config) (v CoordinatorScope, err error) {
-	return newCoordinatorScope(ctx, coordinatorParentScopesImpl{
-		ServiceScope:         svcScope,
-		DistributionScope:    distScp,
-		DistributedLockScope: distLockScp,
-	}, cfg)
+	return newCoordinatorScope(ctx, svcScope, distScp, distLockScp, cfg)
 }
 
-func NewMockedCoordinatorScope(t *testing.T, opts ...dependencies.MockedOption) (CoordinatorScope, Mocked) {
-	t.Helper()
-	return NewMockedCoordinatorScopeWithConfig(t, nil, opts...)
+func NewMockedCoordinatorScope(tb testing.TB, ctx context.Context, opts ...dependencies.MockedOption) (CoordinatorScope, Mocked) {
+	tb.Helper()
+	return NewMockedCoordinatorScopeWithConfig(tb, ctx, nil, opts...)
 }
 
-func NewMockedCoordinatorScopeWithConfig(tb testing.TB, modifyConfig func(*config.Config), opts ...dependencies.MockedOption) (CoordinatorScope, Mocked) {
+func NewMockedCoordinatorScopeWithConfig(tb testing.TB, ctx context.Context, modifyConfig func(*config.Config), opts ...dependencies.MockedOption) (CoordinatorScope, Mocked) {
 	tb.Helper()
-	svcScp, mock := NewMockedServiceScopeWithConfig(
-		tb,
-		modifyConfig,
-		append([]dependencies.MockedOption{dependencies.WithEnabledDistribution("test-node"), dependencies.WithEnabledDistributedLocks()}, opts...)...,
```

Comment:
There was static value `test-node` for the distribution scope.

At the same time, in this part of the code, the config was not yet available so that it was possible to use `cfg.NodeID`.

-----------

It was solved by removing the scopes from the `Mocked` interface, ... the `nodeID` is thus taken from the config, just like in the production setup.